core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2WebUiAuthenticationFilter.java

```diff
@@ -432,4 +431,11 @@ private static void assertRedirectUrl(String redirectUrl)
         void assertWith(WebDriver driver, WebDriverWait wait)
                 throws Exception;
     }
+
+    private static Cookie getCookieNamed(WebDriver driver, String name)
+    {
+        Cookie cookie = driver.manage().getCookieNamed(name);
+        assertThat(cookie).withFailMessage(name + " is missing").isNotNull();
```

Comment:
this could be part of `assertTrinoCookie` and then you don't need the `getCookieNamed` method (which someone may forget to call)

Reply:
Then I would need to modify `assertUICallWithCookie` as well.

I hope that someday @lukasz-walkiewicz will solve the flakiness and this code is not needed.